seqr/migrations/0040_auto_20180603_1309.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11 on 2018-06-03 13:09
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+def transfer_info_from_dataset_to_sample_fields(apps, schema_editor):
+    # since Django doesn't currently support creating new records inside a RunPython migration.
+    # for cases where a sample is attached to multiple datasets (for example both VARIANTS and read alignments),
+    # use the existing Sample record to store metadata from the newest VARIANTS Dataset, and use
+    # update_projects_in_new_schema to create the additional Sample records for read alignments.
+    Dataset = apps.get_model('seqr', 'Dataset')
+    Sample = apps.get_model('seqr', 'Sample')
+
+    print("\nTransferring metadata from {0} Dataset records to {1} Sample records".format(
+        Dataset.objects.filter(analysis_type="VARIANTS").count(),
+        Sample.objects.count(),
+        ))
+
+    for sample in Sample.objects.all():
+        dataset = sample.dataset_set.filter(analysis_type="VARIANTS").order_by('-loaded_date', '-pk').first()
```

Comment:
so this only transfers the most recent dataset right? but don't we also need to keep track of the oldest dataset so we know when it was first loaded?

Reply:
yes. I didn't find a way to create new Sample records during the migration, so left that to update_projects_in_new_schema